include/envoy/http/async_client.h

```diff
@@ -75,6 +75,13 @@ class AsyncClient {
      */
     virtual void onTrailers(HeaderMapPtr&& trailers) PURE;
 
+    /**
+     * Called when both the local and remote have gracefully closed the stream.
+     * Useful for asymmetric cases where end_stream may not be bidirectionally observable.
+     * Note this is NOT called on stream reset.
```

Comment:
Is this true based on the comments below? It seems like we are setting local/remote closed in reset cases?

Reply:
This should still be true.